gobblin-data-management/src/main/java/gobblin/data/management/retention/DatasetCleaner.java

```diff
@@ -53,6 +71,10 @@ public DatasetCleaner(FileSystem fs, Properties props) throws IOException {
     } catch (InvocationTargetException exception) {
       throw new IOException(exception);
     }
+    this.executor =
+        Executors
+            .newFixedThreadPool(Integer.parseInt(props.getProperty(DATASET_CLEAN_THREAD_POOL_SIZE,
```

Comment:
Ws should be using a cached thread pool instead of a fixed one and ideally avoid specifying a thread pool size as this is yet another knob that is hard to tune.

Instead, we should have an option that limits the rate of HDFS operations we perform. Then we should use as many threads are necessary to maintain that rate. @liyinan926 can give you some context as he already has added standard rate limiter implementations.